google/cloud/storage/internal/openssl_util.h

```diff
@@ -0,0 +1,245 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_INTERNAL_OPENSSL_UTIL_H_
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_INTERNAL_OPENSSL_UTIL_H_
+
+#include "google/cloud/internal/throw_delegate.h"
+#include "google/cloud/storage/internal/credential_constants.h"
+#include "google/cloud/storage/version.h"
+#include <openssl/bio.h>
+#include <openssl/buffer.h>
+#include <openssl/evp.h>
+#include <openssl/opensslv.h>
+#include <openssl/pem.h>
+#include <algorithm>
+#include <iostream>
+#include <memory>
+#include <sstream>
+#include <vector>
+
+namespace google {
+namespace cloud {
+namespace storage {
+inline namespace STORAGE_CLIENT_NS {
+namespace internal {
+/**
+ * Helper functions for Base64 and related transcoding.
+ */
+struct OpenSslUtils {
+ public:
```

Comment:
If it is a `struct` then everything is public by default.  In general the practice is to use `struct` for plain old collections of values, and class for things that have member functions (but I am notoriously inconsistent about this).

Reply:
Yeahhh. I made this a struct to signify that it's a container for a bunch of static "util"-type functionality, with no data members or member functions.  These methods are private because they're only used in the actual interesting methods above.  It was between making a struct and a separate namespace for this stuff -- either is fine, but I decided on this over adding another nested namespace.  Thoughts?